frontend/docsettings.lua

```diff
@@ -165,19 +165,112 @@ function DocSettings:open(docfile)
     return setmetatable(new, {__index = DocSettings})
 end
 
---- Reads a setting.
-function DocSettings:readSetting(key)
+--[[-- Reads a setting, optionally initializing it to a default.
+
+If default is set, and the key doesn't exist yet, it is initialized to default first.
+This ensures both that the defaults are actually set if necessary,
+and that the returned reference actually belongs to the DocSettings object straight away,
+without requiring further interaction (e.g., saveSetting) from the caller.
+
+This is mainly useful if the data type you want to retrieve/store is assigned/returned/passed by reference (e.g., a table),
+and you never actually break that reference by assigning another one to the same variable, (by e.g., assigning it a new object).
+c.f., https://www.lua.org/manual/5.1/manual.html#2.2
+
+@param key The setting's key
+@param default Initialization data (Optional)
+]]
+function DocSettings:readSetting(key, default)
+    -- No initialization data: legacy behavior
+    if not default then
+        return self.data[key]
+    end
+
+    if not self:has(key) then
+        self.data[key] = default
+    end
     return self.data[key]
 end
 
 --- Saves a setting.
 function DocSettings:saveSetting(key, value)
     self.data[key] = value
+    return self
 end
 
 --- Deletes a setting.
 function DocSettings:delSetting(key)
     self.data[key] = nil
+    return self
+end
+
+--- Checks if setting exists.
+function DocSettings:has(key)
+    return self.data[key] ~= nil
+end
+
+--- Checks if setting does not exist.
+function DocSettings:hasNot(key)
+    return self.data[key] == nil
+end
+
+--- Checks if setting is `true`.
+function DocSettings:isTrue(key)
+    return string.lower(tostring(self.data[key])) == "true"
```

Comment:
(I have always find these test stranges when re-reading them :/ Does it or did it happen we stored "true" as a string?
There could be edge cases where "true" as a string - a relative path :) - might mean something different than `true` as a boolean.)

Reply:
I asked myself the same question ;).

I'll dig in the history, but I suspect they were written that way from the start for... some reason :?.

Reply:
It appears to -- weirdly -- have been designed for string values from the get-go: https://github.com/koreader/koreader/pull/2215

This doesn't make much sense to me, and I'm inclined to make it purely boolean and what (nothing) break, because storing true as a string is stupid ^^.

Reply:
^ I agree to try that :)

Reply:
Oh, look: the only thing that was doing stupid shit was some tests :}.